gateway/protocol.go

```diff
@@ -82,11 +83,25 @@ type ProtocolRequestContext interface {
 	// and receives and verifies the response.
 	HandleServiceRequest(protocol.Payload) (protocol.Response, error)
 
-	// HandleWebsocketRequest returns a WebsocketsBridge that handles a WebSocket connection request.
-	// A Bridge is returned from this method in order to:
-	//   - Adhere to the convention of the `gateway` package handling non-protocol specific request logic
-	//   - Allow passing gateway level observations to the WebsocketsBridge
-	HandleWebsocketRequest(*http.Request, http.ResponseWriter) (WebsocketsBridge, error)
+	// GetWebsocketConnectionHeaders returns protocol-specific headers needed for websocket connections.
+	// These headers contain protocol-specific information like session data, service IDs, etc.
+	GetWebsocketConnectionHeaders() http.Header
+
+	// GetWebsocketEndpointURL returns the websocket URL for the selected endpoint.
+	// This URL is used to establish the websocket connection to the endpoint.
+	GetWebsocketEndpointURL() (string, error)
+
+	// SignClientWebsocketMessage signs the client message before sending it to the Endpoint on the network.
+	SignClientWebsocketMessage([]byte) ([]byte, error)
+
+	// ValidateEndpointWebsocketMessage validates the websocket message received from the Endpoint on the network.
+	ValidateEndpointWebsocketMessage([]byte) ([]byte, error)
+
+	// UpdateMessageObservationsFromSuccess updates the message observations from a successful message.
+	UpdateMessageObservationsFromSuccess(*observation.RequestResponseObservations) *protocolobservations.Observations
```

Comment:
Great improvement splitting HTTP and Websocket request contexts.
Please revise this interface though: some of these methods are not the responsibility of a protocol, and some of them are internal details of the protocol (e.g. Sign operations).